docs/user/openstack/observability.md

```diff
@@ -0,0 +1,278 @@
+
+There are a number of ways to get your metrics from the same instance. We'll
+document two of them: using Prometheus' remote-write to send the metrics to
+another instance, and using the federation endpoint to query metrics from
+a different instance.
+
+### Using remote-write to send RHOSO and OCP metrics to an external storage
+
+#### Setup external storage
+
+In this example, we'll use an external storage to store the metrics. We'll
+assume that the external storage is another Prometheus instance.
+
+We will set up remote-write from both instances, authenticating them with mTLS
+(mutual TLS). The target Prometheus needs to be configured to accept client TLS
+certificates, and remote write.
+
+// TODO: point at prometheus doc on how to accept TLS certificate and remote write
+
+<!--
+For testing purpose, we can do the following:
+
+1. Provision a Fedora VM
+1. Install `dnf install golang-github-prometheus caddy`
+1. Configure prometheus to enable remote write. In `/etc/default/prometheus`, add the following line:
+```
+ARGS='--enable-feature=remote-write-receiver --storage.tsdb.retention.time=1d'
+```
+1. Start/enable the services
+1. Open security group (443)
+1. Setup caddy, in `/etc/caddy/Caddyfile`:
+
+```Caddyfile
+https://external-prometheus.example {
+    # Just an example
+    basicauth {
+        # caddy hash-password
+        user hashed-password
+    }
+
+    reverse_proxy http://localhost:9090
+}
+```
+
+To generate test certificates:
+
+```bash
+# Generate a CA if you don't have one already
+openssl genrsa -out ca.key 4096
+openssl req -batch -new -x509 -key ca.key -out ca.crt
+
+# Generate the client certificates and sign them:
+for target in server ocp-client osp-client; do
+    openssl genrsa -out "${target}.key" 4096
+    openssl req -batch -new -key "${target}.key" -out "${target}.csr"
+    openssl x509 -req -CA ca.crt -CAkey ca.key -CAcreateserial -in "${target}.csr" -out "${target}.crt"
+done
+```
+
+Prometheus UI should be available at https://external-prometheus.example
+-->
+
+#### Set up remote-write from RHOSO's telemetry-operator
+
+Telemetry should already be enabled in the RHOSO environment. If this not yet
+the case, refer to the
+[documentation](https://docs.redhat.com/en/documentation/red_hat_openstack_services_on_openshift/18.0/html/customizing_the_red_hat_openstack_services_on_openshift_deployment/rhoso-observability_custom_dataplane#rhoso-observability_rhoso-observability).
+
+> [!NOTE]
+Make sure you have the Cluster Observability Operator installed in the OpenShift cluster running the OpenStack control plane, as this is a requirement for the OpenStack Telemetry Operator. Follow [these directions](https://github.com/openstack-k8s-operators/architecture/blob/main/examples/dt/uni01alpha/control-plane.md#cluster-observability-operator) to install it.
+
+To confirm that the Telemetry Operator is correctly running, follow these steps
+
+// TODO: provide steps to ensure telemetry operator is installed and running
```

Comment:
~~~suggestion
To check that the telemetry machinery is correctly installed, you can issue this command:

```bash
oc -n openstack get monitoringstacks metric-storage -o yaml
```

The `monitoringstacks` CRD being installed is a good indicator that telemetry is functional.
~~~